### DISCUSSION TITLE
Test booting the raspbian image

### DISCUSSION THREAD

**Discussion by thk686** [gitbls/sdm#219] [Q&A]
Thanks for the nice tool.

I have written a Dockerfile and driver script that customizes a raspbian image for use in a remote sensing project. I would like to be able to boot the image and test it out without having to burn it to a card and boot a real rpi.

I am trying to use the docker image to run qemu-system-arm with the customized raspbian image. The container starts up and the qemu session runs. However, I cannot see any output. The image should autologin to the console. Has anyone tried this? It would be great if there were an sdm command for booting and testing the image using qemu.

**Comment by gitbls** [comment]
I've not done anything like this with sdm, but it sounds interesting. Happy to help where I can from the sdm perspective. 

Hopefully some other sdm users have more experience with this!

**Comment by thk686** [comment]
Here's the command:
```
docker run -it --rm --privileged -v .:/home/agent/hostdir -v /dev:/dev "$DOCKER_IMAGE_NAME" /bin/bash -v -c "
sudo fdisk -l hostdir/$1
sudo losetup -Pf hostdir/$1
sudo mkdir -p /mnt/raspbian_boot
sudo mkdir -p /mnt/raspbian_root
sudo mount /dev/loop0p1 /mnt/raspbian_boot
sudo mount /dev/loop0p2 /mnt/raspbian_root
sudo qemu-img resize -f raw $CONTAINER_IMAGE_PATH 2G
sudo qemu-system-arm \
  -kernel /mnt/raspbian_boot/kernel.img \
  -dtb /mnt/raspbian_boot/bcm2710-rpi-3-b.dtb \
  -cpu arm1176 \
  -m 1G \
  -machine raspi2b \
  -no-reboot \
  -append 'root=/dev/sda2 rootfstype=ext4 rw' \
  -drive file=$CONTAINER_IMAGE_PATH,format=raw \
  -display curses \
  -nographic
"
```

**Reply by thk686** [reply]
Maybe I can get networking setup and try to ssh in.

**Reply by gitbls** [reply]
I'll have a look at this mid-week when I'm back at my desk. Thx for the command example!

**Comment by gitbls** [comment]
I've spent enough time looking at this to not be interested in solving the basic problem: getting a RasPiOS IMG running in qemu (without docker). I spent a bit of time getting a basic qemu to start, but it took the system compute-bound on all 4 cores. 

So, I'm off onto other projects. But, if you provide a basic script or web site that works, with a reasonable description, I would be more than happy to look into integrating some capability into sdm. IOW it doesn't need to be all bells-and-whistles, but it should just work doing the basic function of running a RasPiOS IMG in qemu.

Thx
